pkg/arvo/app/graph-store.hoon

```diff
@@ -618,392 +615,272 @@
-    =/  =ship  (slav %p i.t.t.path)
-    =/  =term  i.t.t.t.path
-    =/  start=(unit atom)  (rush i.t.t.t.t.path dem:ag)
-    =/  end=(unit atom)    (rush i.t.t.t.t.t.path dem:ag)
-    =/  =index:store
-      (turn t.t.t.t.t.t.path |=(=cord (slav %ud cord)))
-    =/  node=(unit node:store)  (get-node ship term index)
-    ?~  node  [~ ~]
-    ?-  -.children.u.node
-        %empty  [~ ~]
-        %graph
+      [%x %graph @ @ *]
+    =/  =ship   (slav %p i.t.t.path)
+    =/  =term   i.t.t.t.path
+    =/  marked-graph=(unit marked-graph:store)
+      (~(get by graphs) [ship term])
+    ?~  marked-graph  [~ ~]
+    =*  graph  p.u.marked-graph
+    =*  mark   q.u.marked-graph
+    ?+    t.t.t.t.path  (on-peek:def path)
+        ~  
+      :-  ~  :-  ~  :-  %graph-update-2
+      !>(`update:store`[now.bowl [%add-graph [ship term] graph mark %.y]])
+    ::
+        [%mark ~]
+      ``noun+!>(`(unit ^mark)`mark)
+    ::
+        [%subset ?(%lone %kith) @ @ ~]
+      =/  start=(unit atom)  (rush i.t.t.t.t.t.t.path dem:ag)
+      =/  end=(unit atom)    (rush i.t.t.t.t.t.t.t.path dem:ag)
       :-  ~  :-  ~  :-  %graph-update-2
       !>  ^-  update:store
-      :-  now.bowl
-      :+  %add-nodes
-        [ship term]
+      :^  now.bowl  %add-nodes  [ship term]
       %-  ~(gas by *(map index:store node:store))
-      %+  turn  (tap:orm `graph:store`(lot:orm p.children.u.node end start))
+      %+  turn  (tap:orm (lot:orm graph start end))
       |=  [=atom =node:store]
       ^-  [index:store node:store]
-      [(snoc index atom) node]
-    ==
-  ::
-      [%x %deep-nodes-older-than @ @ @ @ ~]
-    =/  =ship  (slav %p i.t.t.path)
-    =/  =term  i.t.t.t.path
-    =/  count=(unit atom)  (rush i.t.t.t.t.path dem:ag)
-    =/  start=(unit atom)  (rush i.t.t.t.t.t.path dem:ag)
-    ?:  ?=(~ count)
-      [~ ~]
-    =/  result=(unit marked-graph:store)
-      (~(get by graphs) [ship term])
-    ?~  result
-      [~ ~]
-    :-  ~  :-  ~  :-  %graph-update-2
-    !>  ^-  update:store
-    :-  now.bowl
-    :+  %add-nodes
-      [ship term]
-    =*  a  u.count
-    =/  b=(list (pair atom node:store))
-      (tab:orm p.u.result start u.count)
-    =|  c=index:store
-    =|  d=(map index:store node:store)
-    =|  e=@ud
-    =-  d
-    |-  ^-  [e=@ud d=(map index:store node:store)]
-    ?:  ?|(?=(~ b) =(e a))
-      [e d]
-    =*  atom  p.i.b
-    =*  node  q.i.b
-    =.  c  (snoc c atom)
-    ?-  -.children.node
-      %empty
-      $(b t.b, e +(e), d (~(put by d) c node), c (snip c))
+      :-  atom^~
+      ?:  ?=(%kith i.t.t.t.t.t.path)
+        node
+      node(children [%empty ~])
     ::
-        %graph
-      =/  f  $(b (tab:orm p.children.node ~ (sub a e)))
-      ?:  =(e.f a)  f
-      %_  $
-        b  t.b
-        e  +(e.f)
-        d  (~(put by d.f) c node(children [%empty ~]))
-        c  (snip c)
+        [%node *]
+      |^
+      =*    pax  t.t.t.t.t.path
+      ?+    pax  (on-peek:def path)
+          [%exists ^]
```

Comment:
Have you considered using the `%u` care for existence checks? That's more generically legible, think namespace exploration and such.

Reply:
Would be neat if `%u` scries were `HEAD` requests